<?xml version="1.0"?>
<!-- launch rosbag and command -->
<launch>

	<rosparam command="load" 	file="$(find panda_controllers)/config/panda_controllers_default.yaml" />

    <!-- flag -->
    <arg name="command_node"    default="true" />
    <arg name="flag_node"       default="false" />
    <arg name="clik"            default="false" />

    <!-- rosbag -->
    <arg name="t1"          value="/backstepping_controller/current_config"/>
    <arg name="t2"          value="/joint_states"/>
    <arg name="t3"          value="/backstepping_controller/command_cartesian"/>
    <arg name="t4"          value="/backstepping_controller/logging"/>
    <arg name="topics"      value="$(arg t1) $(arg t4) $(arg t3) $(arg t2)"/>

    <arg name="bag_path"    default="$(find panda_controllers)/rosbag"/>
    <arg name="bag_name"    default="$(arg bag_path)/prova"/>
    <arg name="duration"    default="30"/>
    
    <!-- rosbag file name -->
    <arg name="gainLambda"  default="L(100.0_1.0)"/>
    <arg name="gainR"       default="Ri1(1_0.1_0.01)"/>
    <arg name="gainKd"      default="Kd(10.0)"/>
    <arg name="adapt"       default="true"/>
    <arg name="traj_type"   default="liss"/> <!-- liss - circ - minjerk -->
    <arg name="param_off"   default="p2"/>  <!-- p0, p2, p5, p10, p20 -->
    <arg name="gain"        default="$(arg gainLambda)_$(arg gainR)_$(arg gainKd)"/>
    
    <node pkg="rosbag" type="record" name="bag_recorder" output="screen" 
        args="--duration=$(arg duration) -O $(arg bag_name)_$(arg traj_type)_$(arg adapt)_$(arg gain)_$(arg param_off) $(arg topics)">
    </node>

    <group ns="backstepping_controller">
        <node pkg="panda_controllers" type="adaptiveFlag_node" name="adaptiveFlag_name" output="screen" if="$(arg flag_node)" />
    </group>

    <include file="$(find panda_controllers)/launch/command.launch" if="$(arg command_node)" >
        <arg name="controller_name" value="backstepping_controller" />
        <arg name="clik"            value="$(arg clik)" />
    </include>

</launch>